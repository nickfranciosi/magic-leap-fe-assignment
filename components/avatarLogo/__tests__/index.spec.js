/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import AvatarLogo from '../index';

describe('AvatarLogo', () => {
  test('it should render', () => {
    const tree = renderer.create(<AvatarLogo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it should accept a size prop', () => {
    const tree = renderer.create(<AvatarLogo size={5} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('width', '5px');
    expect(tree).toHaveStyleRule('height', '5px');
  });

  test('it should accept an avatar prop', () => {
    const tree = renderer
      .create(<AvatarLogo avatar="/path/to/image" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule(
      'background',
      "url('/path/to/image') no-repeat",
      { modifier: ':after' }
    );
  });
});
