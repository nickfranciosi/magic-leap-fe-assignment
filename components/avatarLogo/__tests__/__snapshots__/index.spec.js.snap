// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarLogo it should accept a size prop 1`] = `
.c0 {
  width: 5px;
  height: 5px;
  background: #fff;
  border-radius: 50%;
  overflow: hidden;
  position: relative;
  margin-right: 16px;
  cursor: pointer;
}

.c0:after {
  content: '';
  position: absolute;
  width: 5px;
  height: 5px;
  font-size: 5px;
  bottom: -0.25em;
  background: url('') no-repeat;
  background-size: 100%;
  -webkit-transition: -webkit-transform 350ms ease-in-out;
  -webkit-transition: transform 350ms ease-in-out;
  transition: transform 350ms ease-in-out;
  -webkit-transform: translate(0) scale(1);
  -ms-transform: translate(0) scale(1);
  transform: translate(0) scale(1);
}

.c0:hover:after {
  -webkit-transform: translateY(-.05em) scale(1.03);
  -ms-transform: translateY(-.05em) scale(1.03);
  transform: translateY(-.05em) scale(1.03);
}

@media (max-width:768px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
  size={5}
/>
`;

exports[`AvatarLogo it should accept an avatar prop 1`] = `
.c0 {
  width: px;
  height: px;
  background: #fff;
  border-radius: 50%;
  overflow: hidden;
  position: relative;
  margin-right: 16px;
  cursor: pointer;
}

.c0:after {
  content: '';
  position: absolute;
  width: px;
  height: px;
  font-size: px;
  bottom: -0.25em;
  background: url('/path/to/image') no-repeat;
  background-size: 100%;
  -webkit-transition: -webkit-transform 350ms ease-in-out;
  -webkit-transition: transform 350ms ease-in-out;
  transition: transform 350ms ease-in-out;
  -webkit-transform: translate(0) scale(1);
  -ms-transform: translate(0) scale(1);
  transform: translate(0) scale(1);
}

.c0:hover:after {
  -webkit-transform: translateY(-.05em) scale(1.03);
  -ms-transform: translateY(-.05em) scale(1.03);
  transform: translateY(-.05em) scale(1.03);
}

@media (max-width:768px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
/>
`;

exports[`AvatarLogo it should render 1`] = `
.c0 {
  width: px;
  height: px;
  background: #fff;
  border-radius: 50%;
  overflow: hidden;
  position: relative;
  margin-right: 16px;
  cursor: pointer;
}

.c0:after {
  content: '';
  position: absolute;
  width: px;
  height: px;
  font-size: px;
  bottom: -0.25em;
  background: url('') no-repeat;
  background-size: 100%;
  -webkit-transition: -webkit-transform 350ms ease-in-out;
  -webkit-transition: transform 350ms ease-in-out;
  transition: transform 350ms ease-in-out;
  -webkit-transform: translate(0) scale(1);
  -ms-transform: translate(0) scale(1);
  transform: translate(0) scale(1);
}

.c0:hover:after {
  -webkit-transform: translateY(-.05em) scale(1.03);
  -ms-transform: translateY(-.05em) scale(1.03);
  transform: translateY(-.05em) scale(1.03);
}

@media (max-width:768px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
/>
`;
